### PR TITLE
Namespace/Docstring Fixes for Reduction

### PR DESCRIPTION
**Pull request by isVoid**
This PR adds detail namespace for for simple/compound ops for reduction to make code more consistent with segmented reduction. It also includes minor docstring fixes.

cc @bdice

### CODE REVIEW

File: cpp/tests/reductions/segmented_reduction_tests.cpp [hunk 1]
```diff
@@ -15,10 +15,12 @@
  */
 
 #include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_utilities.hpp>
```
Comment (bdice) [inline cpp/tests/reductions/segmented_reduction_tests.cpp:18]: @isVoid Can you revert this file? (It's meant to be part of another PR.)

File: cpp/tests/reductions/segmented_reduction_tests.cpp [hunk 2]
```diff
@@ -385,6 +387,91 @@ TEST_F(SegmentedReductionTestUntyped, ReduceEmptyColumn)
...
+                                {true, false, true, false, false, false}};
+
+  auto res = segmented_reduce(input,
+                              column_view(offsets),
+                              *make_max_aggregation<segmented_reduce_aggregation>(),
+                              output_dtype,
+                              null_policy::INCLUDE);
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(*res, expect);
+}
+
```
Comment (isVoid) [inline cpp/tests/reductions/segmented_reduction_tests.cpp:474]:
```suggestion
```

**Review by isVoid — COMMENTED** [review]
Things carried from another PR.

File: cpp/tests/reductions/segmented_reduction_tests.cpp (hunk 1 above)
Comment (isVoid) [reply cpp/tests/reductions/segmented_reduction_tests.cpp:18]: Yeah, just about to do that.

**Comment by bdice** [comment]
This is breaking because it changes namespaces by moving internal code into a `detail` namespace. I added the `breaking` label.

File: cpp/include/cudf/detail/reduction_functions.hpp [hunk 3]
```diff
@@ -192,6 +192,8 @@ std::unique_ptr<scalar> mean(
  *
  * @param col input column to compute variance.
  * @param output_dtype data type of return type and typecast elements of input column.
+ * @param ddof Delta degrees of freedom. The divisor used is N - ddof, where N represents the number
+ * of elements.
```
Comment (hyperbolic2346) [inline cpp/include/cudf/detail/reduction_functions.hpp:196]: 👍

**Review by hyperbolic2346 — APPROVED** [review]
LGTM

**Comment by bdice** [comment]
rerun tests

**Review by bdice — APPROVED** [review]

**Comment by isVoid** [comment]
@gpucibot merge
